pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/ServerClientAnnotator.java

```diff
@@ -82,7 +82,7 @@ public void useCaching() {
     }
 
     public void useCaching(String dbFile) {
-        this.db = DBMaker.fileDB(dbFile).closeOnJvmShutdown().make();
+        this.db = DBMaker.fileDB(dbFile).closeOnJvmShutdown().transactionEnable().make();
```

Comment:
I think you will also have to call `this.db.commit()` whenever you make an update to the data. 

https://jankotek.gitbooks.io/mapdb/content/db#transactions